game/source_engine_nav.ksy

```diff
@@ -0,0 +1,467 @@
+        doc: The north-west corner position of an area.
+      - id: south_east_corner
+        type: f4
+        repeat: expr
+        repeat-expr: 3
+        doc: The south-east corner position of an area.
+      - id: north_east_corner_height
+        type: f4
+        doc: The height (Z coordinate) of the area's north-east corner.
+      - id: south_west_corner_height
+        type: f4
+        doc: The height (Z coordinate) of the area's south-west corner.
+      # Connection data.
+      - id: connection_data
+        type: area_id_array
+        repeat: expr
+        repeat-expr: 4
+        doc: Container for connections.
+      # Hiding Spots sequence.
+      - id: hiding_spot_count
+        type: u1
+        doc: Amount of hiding spots stored in the navigation area.
+      - id: hiding_spot_sequence
+        type: hiding_spot
+        repeat: expr
+        repeat-expr: hiding_spot_count
+        doc: Sequence of hiding spots.
+      # Approach Spots
+      - id: approach_info_count
+        type: u1
+        if: '_root.version < 15'
+        doc: |
+          The amount of approach spots stored in the NAV area.
+
+          Removed in NAV version 15 and above.
+      - id: approach_info_sequence
+        type: approach_info
+        repeat: expr
+        repeat-expr: approach_info_count
+        if: '_root.version < 15'
+      # Encounter Path
+      - id: encounter_path_count
+        type: u4
+        doc: Amount of encounter paths.
+      - id: encounter_path_data
+        type: encounter_path
+        repeat: expr
+        repeat-expr: encounter_path_count
+      # Place ID
+      - id: place_id
+        type: u2
+        doc: Reference place index.
+      # Ladders
+      - id: ladder_data
+        type: area_id_array
+        repeat: expr
+        repeat-expr: 2
+      # Occupy times
+      - id: earliest_occupation_time
+        type: f4
+        repeat: expr
+        repeat-expr: 2
+        if: '_root.version >= 8'
+        doc: Earliest occupation times possible for each team. Added in NAV version 8.
+      # Light Intensity
+      - id: light_intensity
+        type: f4
+        repeat: expr
+        repeat-expr: 4
+        if: '_root.version >= 11'
+        doc: Light Intensities (brightness) for each corner. Added in NAV version 11.
+      # Visibility area-binds.
+      - id: area_bind_count
+        type: u4
+        if: '_root.version >= 16'
+        doc: Amount of area-binds stored in area data. Added in NAV version 16.
+      - id: area_bind_array
+        type: area_bind
+        repeat: expr
+        repeat-expr: area_bind_count
+        if: '_root.version >= 16'
+      # Inheritance
+      - id: area_id_vis_inherit
+        type: u4
+        if: '_root.version >= 16'
+        doc: ID of area to inherit visibility bind information from. Only exists in NAV version 16 and later.
+      # Custom data can be appended here.
+    enums:
+     # Enum in code: NavAttributeType
+      nav_attribute_type:
```

Comment:
how about converting this one? `bit-endian: le` can be helpful.

Reply:
What do you mean by convert?

Reply:
Create a type, put individual bit fields as `b1` into it, as you have done with other enums defining flags. But note, if you enable `bit-endian: le`, the order of bit-fields in `seq` must be reversed relatively to the order used by default in KS (I don't approve the decision to implement that feature in KSC that way as it violates the convention MSBs are listed first in `seq` which interferes with switcheable endianness, I had an alternative proposal). This means that you keep the order currently used in the enum.

Reply:
Ok I see what you mean. I did what you mentioned above.

It seems that the hide spot attribute flag is written with little bit-endian, so since I used `bit-endian: le`, endianess shouldn't be causing too much problems.

Reply:
I converted `nav_attribute_type` to a variable-lenth bit field.